### PR TITLE
Add MacOS support to cibuildwheels

### PR DESCRIPTION
**Pull request by whytro**
I've shifted the cibuildwheel configuration from the pyproject.toml to the workflow files, to take advantage of GHA's feature to view the Workflow file that was run.

I've also had to take care of some code issues (ie. missing includes) that prevented compilation on MacOS, though I will take a closer look at the code soon - to make sure there aren't any more missing includes, or formatting issues.

Currently, the time to run the osrm-backend workflow is around 7-9 minutes, while the time to build the MacOS wheels seems to be around 8-9 minutes. Without separation, its about 12 minutes. There is maybe room for more optimization here, since most of the time seems to be sunk into installing the dependencies like `brew install lua tbb boost@1.76`, but I wonder if it is faster/feasible to pack those into the artifact.

### CODE REVIEW

File: .github/workflows/osrm-backend.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      with:
+        repository: Project-OSRM/osrm-backend
+        submodules: true
+        fetch-depth: 1
+        path: osrm-backend
+
+    - name: Build osrm-backend
+      run: |
+        brew install lua tbb
+        brew install boost@1.76 && brew link boost@1.76
```
Comment (nilsnolde) [inline .github/workflows/osrm-backend.yml:30]: is there a reason why we pin boost to 1.76?

Comment (SiarheiFedartsou) [reply .github/workflows/osrm-backend.yml:30]: I think so, AFAIR there is a bug in the latest boost which doesn't allow OSRM to build.

File: .github/workflows/osrm-backend.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      uses: actions/checkout@v3
+      with:
+        repository: Project-OSRM/osrm-backend
+        submodules: true
+        fetch-depth: 1
+        path: osrm-backend
+
+    - name: Build osrm-backend
+      run: |
+        brew install lua tbb
```
Comment (SiarheiFedartsou) [inline .github/workflows/osrm-backend.yml:29]: setting `HOMEBREW_NO_AUTO_UPDATE=1` can potentially speed it up

**Review by SiarheiFedartsou — APPROVED** [review]

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -4,12 +4,56 @@ on: [pull_request]
...
       uses: pypa/cibuildwheel@v2.14.1
+      env:
+        CIBW_BUILD_VERBOSITY: 1
+
+        CIBW_TEST_REQUIRES: pytest
+        CIBW_TEST_COMMAND: "pytest {project}/tests"
+        CIBW_BEFORE_TEST: osrm-datastore {package}/tests/test_data/ch/monaco
+
+        CIBW_ARCHS: native
+        CIBW_BUILD: cp*
```
Comment (nilsnolde) [inline .github/workflows/pull_request.yml:44]: let's just build the lowest python version, python3.8

File: .github/workflows/osrm-backend.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        version: [latest]
+    env:
+      GITHUB_TOKEN: ${{ github.token }}
+      ARTIFACT_WORKFLOW: osrm-backend.yml
+
+    steps:
+    - name: Checkout osrm-backend
+      uses: actions/checkout@v3
+      with:
+        repository: Project-OSRM/osrm-backend
```
Comment (nilsnolde) [inline .github/workflows/osrm-backend.yml:22]: we should pin osrm to the most recent tag/release

File: .github/workflows/osrm-backend.yml (hunk 1 above)
Comment (whytro) [reply .github/workflows/osrm-backend.yml:30]: Can confirm. There was a duplicate symbols issue that I encountered previously on latest, and brew didn't have 1.79 - hence the 1.76 pin.

**Comment by whytro** [comment]
I've made the requested changes, but the osrm-backend workflow needs to be run before the MacOS workflow in order for that to work.

**Comment by nilsnolde** [comment]
Urgh, right, that's not in master yet. @whytro do you want to break out that workflow file separately into a PR and we'll merge that first, so we can build the artifacts?

**Comment by whytro** [comment]
> Urgh, right, that's not in master yet. @whytro do you want to break out that workflow file separately into a PR and we'll merge that first, so we can build the artifacts?

Sure, I've opened up another PR #15 .

**Review by nilsnolde — APPROVED** [review]
Looks good thanks!

**Pull request merged**